locust/contrib/fasthttp.py

```diff
@@ -75,13 +75,18 @@ def __init__(
         user: Optional[User],
         insecure=True,
         client_pool: Optional[HTTPClientPool] = None,
+        ssl_context: Optional[SSLContext] = None,
         **kwargs,
     ):
         self.environment = environment
         self.base_url = base_url
         self.cookiejar = CookieJar()
         self.user = user
-        if insecure:
+        if ssl_context:
+            if not isinstance(ssl_context, SSLContext):
+                raise TypeError("You must provide a valid SSLContext.")
+            ssl_context_factory = lambda: ssl_context
```

Comment:
This looks a little strange to me. Doesnt it make more sense to just pass the function instead of a value that we then make into a function anyway? I guess we can't check the type as neatly at runtime, but I think it is unnecessary (especially with a type hint it to indicate it must return an SSLContext)